electrumsv/gui/qt/network_dialog.py

```diff
@@ -1288,9 +1291,15 @@ def _on_menu_delete_server(self, entry: ServerListEntry) -> None:
                 for account_id in server.get_account_ids():
                     delete_server_keys.append(ServerAccountKey(entry.url,
                         entry.server_type, account_id))
-                future = self._wallet_weakref.update_network_servers([], [], delete_server_keys, {})
-                future.result()
 
+                # Synchronization is required for the callback deleting cached server data,
```

Comment:
This is a workaround, not the correct fix. What I do know about the correct fix is that:
* We do not want to be blocking the UI thread for 5 seconds.
* We want to do the update in the UI thread (as all UI calls have to be) and there is already a mechanism there that does not seem to be used `server_disconnected_signal`.
* We want `server_disconnected_signal.emit` to be called after the embedded callback in `update_network_servers`, which will run the update call in the UI thread.

The simplest solution would possibly be to have:
```
future = ...
future.add_done_callback(self.server_disconnected_signal.emit)
future.result()
```

We should be able to rely on it being run after the first callback. Maybe the `...emit` call needs to be wrapped in a local function to properly receive the callback too. 

Make sense?

Reply:
5 seconds is only the worst case timeout but yes makes sense.

Will give that a go :) - Might have to be tomorrow.

Reply:
That has worked👍Please squash the commits when you merge this.